lib/storage/metadata/mongoclient/MongoClientInterface.js

```diff
@@ -1035,11 +1034,144 @@ class MongoClientInterface {
+                log.error('could not get list of collections', {
+                    method: '_getBucketInfos',
+                    error: err,
+                });
+                return cb(err);
+            }
+            return async.eachLimit(collInfos, 10, (value, next) => {
+                if (value.name === METASTORE ||
+                    value.name === INFOSTORE ||
+                    value.name === USERSBUCKET ||
+                    value.name === PENSIEVE ||
+                    value.name.startsWith(constants.mpuBucketPrefix)
+                ) {
+                    // skip
+                    return next();
+                }
+                bucketCount++;
+                const bucketName = value.name;
+                // FIXME: there is currently no way of distinguishing
+                // master from versions and searching for VID_SEP
+                // does not work because there cannot be null bytes
+                // in $regex
+
+                return this.getBucketAttributes(bucketName, log,
+                    (err, bucketInfo) => {
+                        if (err) {
+                            return next(errors.InternalError);
+                        }
+                        bucketInfos.push(bucketInfo);
+                        return next();
+                    });
+            }, err => {
+                if (err) {
+                    return cb(err);
+                }
+                return cb(null, {
+                    bucketCount,
+                    bucketInfos,
+                });
+            });
+        });
+    }
+
     countItems(log, cb) {
-        const doFullScan = this.lastItemScanTime === null ||
-            (Date.now() - this.lastItemScanTime) > itemScanRefreshDelay;
+        this._getBucketInfos(log, (err, res) => {
+            if (err) {
+                log.error('error getting bucket info', {
+                    method: 'countItems',
+                    error: err,
+                });
+                return cb(err);
+            }
+            const { bucketCount, bucketInfos } = res;
+
+            const retBucketInfos = bucketInfos.map(bucket => ({
+                name: bucket.getName(),
+                location: bucket.getLocationConstraint(),
+                isVersioned: !!bucket.getVersioningConfiguration(),
+                ownerCanonicalId: bucket.getOwner(),
+                ingestion: bucket.isIngestionBucket(),
+            }));
+
+            return this.readCountItems(log, (err, results) => {
+                if (err) {
+                    return cb(err);
+                }
+                // overwrite bucket info since we have latest info
+                /* eslint-disable */
+                results.bucketList = retBucketInfos;
+                results.buckets = bucketCount;
+                /* eslint-enable */
+                return cb(null, results);
+            });
+        });
+    }
 
-        const res = {
+    scanItemCount(log, cb) {
```

Comment:
minor: since the scanning is moved to a kubernetes job, `scanItemsCount` and the `this.config` usage can be removed. (can be a second PR)

Reply:
I kept the scan function in the class and the s3util script will just require the class and invoke this function. I decided to keep the logic in this class to utilize available methods and vars

Ty for the review, I've removed `this.config` and data counter